### PR TITLE
[UNO-152] The system does not reject decimal entering into the mix/max fields

### PR DESCRIPTION
**Pull request by atsymuk**
Related to : https://oat-sa.atlassian.net/browse/UNO-152
 
Add possibility to disable decimal numbers for minMax component. enable it by default
  
#### How to test
 
- open authoring for any test item
- add choice interaction
- try to change min or max choices
- try to provide decimal value
- check that it converted to number after input delay

### CODE REVIEW

File: views/js/qtiCreator/widgets/component/minMax/minMax.js [hunk 1]
```diff
@@ -350,6 +353,27 @@ define([
...
+             * @param {String} [fromField = min] - min or max, where the change comes from to update accordingly
+             * @returns {minMax} chains
+             * @throws {TypeError} if the field is unknown
+             */
+            convertToNumber: function syncValues(fromField){
+                fromField = fromField || fields.min;
+
+                if (isFieldSupported(fromField) && this.is('rendered')) {
+                    if(fromField === fields.max){
+                        this.setMaxValue(_.parseInt(this.getMaxValue()));
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/component/minMax/minMax.js:368]: Can we use native `parseInt `?

File: views/js/qtiCreator/widgets/component/minMax/minMax.js [hunk 2]
```diff
@@ -350,6 +353,27 @@ define([
...
             },
 
+            /**
+             * convert value to number
+             *
+             * @param {String} [fromField = min] - min or max, where the change comes from to update accordingly
+             * @returns {minMax} chains
+             * @throws {TypeError} if the field is unknown
+             */
+            convertToNumber: function syncValues(fromField){
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/component/minMax/minMax.js:363]:
```suggestion
            convertToNumber: function convertToNumber(fromField){
```

File: views/js/qtiCreator/widgets/component/minMax/minMax.js [hunk 3]
```diff
@@ -350,6 +353,27 @@ define([
...
+                fromField = fromField || fields.min;
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/component/minMax/minMax.js:364]: Why if no `fromField ` will be fields.min?

**Review by bziondik — CHANGES_REQUESTED** [review]
Works for me. Only some comments.

File: views/js/qtiCreator/widgets/component/minMax/minMax.js [hunk 4]
```diff
@@ -177,7 +180,7 @@ define([
                 if ( isFieldSupported(field) && _.isNumber(intValue) &&
                      intValue >= config.lowerThreshold && intValue <= config.upperThreshold ) {
 
-                    if ( this.is('rendered') && controls[field].input.val() !== intValue ) {
+                    if ( this.is('rendered') && parseInt(controls[field].input.val()) !== intValue ) {
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/component/minMax/minMax.js:183]: in case `allowDecimal: true` we need to do parseFloat?

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

File: views/js/qtiCreator/widgets/component/minMax/minMax.js (hunk 4 above)
Comment (atsymuk) [reply views/js/qtiCreator/widgets/component/minMax/minMax.js]: we can use `parseFloat` for both cases

**Review by lecosson — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Pull request merged**